deprecate.dd

```diff
@@ -57,6 +58,36 @@ $(SPEC_S Deprecated Features,
     )
 
 
+$(H3 $(DEPNAME Implicit string concatenation))
+    $(P Currently, two adjacent strings are implicitly concatenated:
+
+        ---
+        string foo = "Hello" "World";
+        ---
+
+        This feature is handy for long string that spans multiple lines,
+        however it is possible to get the same behaviour explicitly by
+        using the concatenation operator ('~'):
+
+        ---
+        string foo = "Hello" ~ "World"; // No allocation is performed
+        ---
+    )
+$(H4 Corrective Action)
+    $(P Replace implicit string concatenation by explicit one, using '~'.)
+$(H4 Rationale)
+    $(P This is a very early feature of the language, which is nowadays totally
+        covered by the concatenation operator: it is performed at compile time
+        for constants and doesn't result in memory allocation.)
+
+    $(P However, having implicit concatenation can and did result in hard
+        to spot bugs, for example:)
+
+        ---
+        string[] arr = [ "Hello", "buggy" "World" ];
```

Comment:
I'd use a more realistic example. The classic mistake is made when the list of strings is separated on each line:

```D
string[] names =
[
    "Anna",
    "Michael"
    "Emma",
    "David"
];
```